app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Error;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Response;
+use Illuminate\Support\Facades\Storage;
+use Mpdf\Mpdf;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        $users = User::all();
+        return view('dashboard.user', ['users' => $users]);
+    }
+
+
+    public function draw(Request $request)
+    {
+        $search = $request->query('search')['value'];
+        $draw = $request->query('draw', 1);
+        $start = $request->query('start', 0);
+        $length = $request->query('length', 10);
+        $usersQuery = User::query();
+        $totalusers = $usersQuery->count();
+
+        if (!$request->dropdowns) {
+            $users = $usersQuery->where('name', 'like', "%" . $search . "%")
+                ->orWhere('email', 'like', "%" . $search . "%")
+                ->orWhere('is_active', 'like', "%" . $search . "%");
+        } else {
+            $searchUser = $request->dropdowns['user'] ?? null;
+            $users = $usersQuery->whereIn('user', $searchUser);
+        }
+
+        if ($order = $request->query('order')[0] ?? null) {
+            $orderDir = $order['dir'] ?? 'asc';
+            $users->orderBy($order['name'], $orderDir);
+        }
+
+        $filteredusers = $search ? $users->count() : $totalusers;
+        $users = $users->skip($start)
+            ->take($length)
+            ->get();
+
+        return Response::json([
+            'draw' => intval($draw),
+            'recordsTotal' => intval($totalusers),
+            'recordsFiltered' => $filteredusers,
+            'data' => $users
+        ]);
+    }
+
+    public function createPDF(){
+        $document = new Mpdf(['orientation' => 'L']);
+        $users = User::all();
+        $html = view('docs.userPdf', ['users' => $users , 'title' => 'User Mangement']);
+        $document->WriteHTML($html);
+        Storage::put('pdfs/user_information.pdf', $document->Output('user_information.pdf', "S"));
```

Comment:
there is no point of storing the PDF file. lets assume you download the PDF and then some one changes the information of the user. so when next we download we have to download from latest information. so no use of storing the PDF file. make sure once the function is triggered it should automatically download the PDF

Reply:
Yes understood...👍